src/integrationTest/java/uk/gov/hmcts/reform/civil/controllers/cases/DocumentControllerTest.java

```diff
@@ -178,16 +189,28 @@ void shouldDownloadDocumentFromDocumentManagement_FromCaseDocumentClientApi() th
             .getDocumentBinary(anyString(), anyString(), eq(documentId));
     }
 
+    @Test
+    void shouldReturnExpectedGeneratedDocument() throws Exception {
+        CaseData caseData = CaseDataBuilder.builder().atStateClaimSubmitted()
+            .legacyCaseReference(REFERENCE_NUMBER)
+            .totalClaimAmount(BigDecimal.ONE)
+            .issueDate(DATE)
+            .build();
+        when(restTemplate.exchange(anyString(), eq(HttpMethod.POST), any(), eq(byte[].class)))
+            .thenReturn(ResponseEntity.of(Optional.of(bytes)));
```

Comment:
You can use BDDMockito here.

```suggestion
        given(restTemplate.exchange(anyString(), eq(HttpMethod.POST), any(), eq(byte[].class)))
            .willReturn(ResponseEntity.of(Optional.of(bytes)));
```

 Make sure to add the right import along with this suggestion.